### PR TITLE
Create the domain upsell steps component

### LINKED ISSUE #72892: Sidebar nudge upsell: Implement the navigation header
[Based on our POC](https://github.com/Automattic/wp-calypso/pull/72831), we expect:

- [x] Implement the new header for the flow
- [x] At the domain page, `Select a Domain` should be selected
- [x] At the plans page, `Select a plan` should be selected
- [x] Back button should go back to the previous page user was
- [ ] Gate the component with `?domainAndPlanPackage=true` and experiment = `treatment`
- [x] It should look like the design: pebzTe-Fx-p2#comment-1131
- [ ] Make sure the changes are connected to the experiment and only show when the user is in the `treatment` group. [Example of code here](https://github.com/Automattic/wp-calypso/pull/72831/commits/8148a79a089580437ddc43cbaa27d6ad03f381e0).

### Design reference

header

task-related: https://github.com/Automattic/wp-calypso/issues/72516

### PR DESCRIPTION
**Pull request by DustyReagan**
#### Proposed Changes

* This PR adds a new steps navigation component for our domain upsell flow.

Step 1 example:
step1

Step 2 example:
step2

#### Testing Instructions

This PR is based on [Prepare the domain page to work with the experiment - backend strategy #73017 ](https://github.com/Automattic/wp-calypso/pull/73017)

* To test this component follow the testing instruction in [Prepare the domain page to work with the experiment - backend strategy #73017 ](https://github.com/Automattic/wp-calypso/pull/73017)
* You should see the navigation header at the top of the Search page. (As seen in the pictures above.)
* To test step 2, modify the component in code to `<DomainAndPlanPackageNavigation step={ 2 } />` and reload the search page. (eventually we'll add this component to the plans page as "step 2").
* Test in different browsers and on mobile.
* Test the "Back" button goes back.

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

### CODE REVIEW

File: client/assets/images/icons/wordpress-logo.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg viewBox="0 0 72 72" xmlns="http://www.w3.org/2000/svg">
+<path d="M36,0C16.1,0,0,16.1,0,36c0,19.9,16.1,36,36,36c19.9,0,36-16.2,36-36C72,16.1,55.8,0,36,0z M3.6,36 c0-4.7,1-9.1,2.8-13.2l15.4,42.3C11.1,59.9,3.6,48.8,3.6,36z M36,68.4c-3.2,0-6.2-0.5-9.1-1.3l9.7-28.2l9.9,27.3 c0.1,0.2,0.1,0.3,0.2,0.4C43.4,67.7,39.8,68.4,36,68.4z M40.5,20.8c1.9-0.1,3.7-0.3,3.7-0.3c1.7-0.2,1.5-2.8-0.2-2.7 c0,0-5.2,0.4-8.6,0.4c-3.2,0-8.5-0.4-8.5-0.4c-1.7-0.1-2,2.6-0.2,2.7c0,0,1.7,0.2,3.4,0.3l5,13.8L28,55.9L16.2,20.8 c2-0.1,3.7-0.3,3.7-0.3c1.7-0.2,1.5-2.8-0.2-2.7c0,0-5.2,0.4-8.6,0.4c-0.6,0-1.3,0-2.1,0C14.7,9.4,24.7,3.6,36,3.6 c8.4,0,16.1,3.2,21.9,8.5c-0.1,0-0.3,0-0.4,0c-3.2,0-5.4,2.8-5.4,5.7c0,2.7,1.5,4.9,3.2,7.6c1.2,2.2,2.7,4.9,2.7,8.9 c0,2.8-0.8,6.3-2.5,10.5l-3.2,10.8L40.5,20.8z M52.3,64l9.9-28.6c1.8-4.6,2.5-8.3,2.5-11.6c0-1.2-0.1-2.3-0.2-3.3 c2.5,4.6,4,9.9,4,15.5C68.4,47.9,61.9,58.4,52.3,64z" />
+</svg>
```
Comment (DustyReagan) [inline client/assets/images/icons/wordpress-logo.svg:3]: A wordpress logo component exists, but I wanted to use the logo as a background in CSS, so I added it as a SVG file.

**Review by paulopmt1 — COMMENTED** [review]
Hey @DustyReagan sorry for the rebase needed due to the changes [to our backend strategy](https://github.com/Automattic/wp-calypso/pull/73017).

After adding the `<DomainAndPlanPackageNavigation step={ 1 } />` [inside the domain-search__conternt](https://github.com/Automattic/wp-calypso/blob/add/domain-upsell-header-component/client/my-sites/domains/domain-search/index.jsx#L238), hiding the top and side menus, and removing padding of `id="content"` div, I was able to test the feature:

image

On small resolutions, it seems we're intentionally hiding them. Should we review with the design team the plans for the small/mobile version of this screen?

image

**Comment by DustyReagan** [comment]
> sorry for the rebase needed due to the changes [to our backend strategy](https://github.com/Automattic/wp-calypso/pull/73017).

No problem! Thanks for working through that!

> On small resolutions, it seems we're intentionally hiding them. Should we review with the design team the plans for the small/mobile version of this screen?

Summarizing our Slack conversation. :) p1675796005776599-slack-CKZHG0QCR It is an intentional judgement call I made on how to handle smaller screens. If there's no objections, we can use this strategy for now and iterate if we get more feedback from design.

**Review by paulopmt1 — APPROVED** [review]
More tests will be needed once we connect this component to the domain and plans page, but based on the tests I was able to do, it's working great. Good job @DustyReagan!

**Comment by JanaMW27** [comment]
Can we do something like:

Screenshot 2023-02-08 at 11 46 23

To keep it simple but visible at the same time.

**Pull request merged**